store/store_action.go

```diff
@@ -385,30 +400,76 @@ func (s *Store) snapAction(ctx context.Context, currentSnaps []*CurrentSnap, act
 
 		aJSON.InstanceKey = instanceKey
 
-		actionJSONs[i] = aJSON
+		actionJSONs[actionIndex] = aJSON
+		actionIndex++
 	}
 
 	// assertions
 	var assertMaxFormats map[string]int
 	if len(toResolve) > 0 {
-		i := len(actionJSONs) - len(toResolve)
 		for grp, ats := range toResolve {
 			aJSON := &snapActionJSON{
 				Action: "fetch-assertions",
 				Key:    string(grp),
 			}
-			aJSON.Assertions = make([]assertAtJSON, len(ats))
+			aJSON.Assertions = make([]interface{}, len(ats))
 			for j, at := range ats {
-				aJSON.Assertions[j].Type = at.Type.Name
-				aJSON.Assertions[j].PrimaryKey = at.PrimaryKey
+				aj := &assertAtJSON{
+					Type:       at.Type.Name,
+					PrimaryKey: at.PrimaryKey,
+				}
 				rev := at.Revision
 				if rev != asserts.RevisionNotKnown {
-					aJSON.Assertions[j].IfNewerThan = &rev
+					aj.IfNewerThan = &rev
 				}
+				aJSON.Assertions[j] = aj
 			}
-			actionJSONs[i] = aJSON
-			i++
+			actionJSONs[actionIndex] = aJSON
+			actionIndex++
 		}
+	}
+
+	if len(toResolveSeq) > 0 {
+		for grp, ats := range toResolveSeq {
+			aJSON := &snapActionJSON{
+				Action: "fetch-assertions",
+				Key:    string(grp),
```

Comment:
I think I mentioned this but I think logically we can have the same grouping appearing in both toResolve and toResolveSeq, in which case we need to produce one fetch-assertions that covers both, this code seems it would produce two different ones with the same key which I don't think the store supports

Reply:
Thanks, I missed that case, fixed now.